types/yargs/index.d.ts

```diff
@@ -44,8 +44,7 @@ declare namespace yargs {
      * `Arguments<T>` to simplify the inferred type signature in client code.
      */
     interface Argv<T = {}> {
-        (): { [key in keyof Arguments<T> as key | CamelCaseKey<key>]: Arguments<T>[key] } | Promise<{ [key in keyof Arguments<T> as key | CamelCaseKey<key>]: Arguments<T>[key] }>;
```

Comment:
Deleting this line implies that using `Arguments<T>` is incorrect for the no-argument case. Is that correct? I don't know yargs well, but this seems like a big change.

Reply:
Let me look a little bit further -- I need to make sure I chose the correct one between `Argv<T>` and `{ [key in keyof Arguments<T> as key | CamelCaseKey<key>]: Arguments<T>[key] }`

Things I know:
- the return type should not be a promise
- `yargs()`, `yargs('hello')`, `yargs(['hello'])` should all have the same generic return type. (I still need to determine which one of the above it is)